File: backend/schemas/client_portal.py
import uuid
from datetime import date, datetime
from typing import Literal, Optional
from pydantic import BaseModel, Field


class ClientProfileOut(BaseModel):
    name: str
    email: str
    phone: Optional[str]


class ClientProfileUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1)
    phone: Optional[str] = None


class ClientJobStageOut(BaseModel):
    id: uuid.UUID
    name: str
    color: str
    position: int


class ClientJobOut(BaseModel):
    id: uuid.UUID
    delivery_url: Optional[str]
    appointment_title: str
    appointment_starts_at: str
    stage_name: str
    stage_color: str


class ClientJobDetailOut(BaseModel):
    id: uuid.UUID
    delivery_url: Optional[str]
    appointment_title: str
    appointment_starts_at: str
    appointment_session_types: list[str]
    stage_id: uuid.UUID
    stage_name: str
    all_stages: list[ClientJobStageOut]


class SessionTypeOut(BaseModel):
    id: uuid.UUID
    name: str
    available_days: list[int]


class ClientBookingRequestSlot(BaseModel):
    session_type_id: uuid.UUID
    session_type_name: Optional[str]
    date: date
    time_slot: Literal["morning", "afternoon", "evening", "all_day"]


class ClientBookingRequestOut(BaseModel):
    id: uuid.UUID
    preferred_date: date
    session_slots: list[ClientBookingRequestSlot]
    # Legacy fields — kept for backwards compat, not written by new code
    time_slot: Optional[Literal["morning", "afternoon", "evening", "all_day"]]
    session_type_name: Optional[str]
    message: Optional[str]
    status: Literal["pending", "confirmed", "rejected"]
    admin_notes: Optional[str]
    created_at: datetime


class ClientBookingRequestSlotCreate(BaseModel):
    session_type_id: uuid.UUID
    date: date
    time_slot: Literal["morning", "afternoon", "evening", "all_day"]


class ClientBookingRequestCreate(BaseModel):
    session_slots: list[ClientBookingRequestSlotCreate]
    message: Optional[str] = None

File: frontend/src/schemas/clientPortal.ts
// frontend/src/schemas/clientPortal.ts
import { z } from 'zod'

export const ClientProfileSchema = z.object({
  name: z.string(),
  email: z.string(),
  phone: z.string().nullable(),
  avatar_url: z.string().nullable().optional(),
})
export type ClientProfile = z.infer<typeof ClientProfileSchema>

export const ClientJobSchema = z.object({
  id: z.string().uuid(),
  delivery_url: z.string().nullable(),
  appointment_title: z.string(),
  appointment_starts_at: z.string(),
  stage_name: z.string(),
  stage_color: z.string(),
})
export type ClientJob = z.infer<typeof ClientJobSchema>
export const ClientJobListSchema = z.array(ClientJobSchema)

export const ClientJobStageSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
  color: z.string(),
  position: z.number().int(),
})
export type ClientJobStage = z.infer<typeof ClientJobStageSchema>

export const ClientJobDetailSchema = z.object({
  id: z.string().uuid(),
  delivery_url: z.string().nullable(),
  appointment_title: z.string(),
  appointment_starts_at: z.string(),
  appointment_session_types: z.array(z.string()),
  stage_id: z.string().uuid(),
  stage_name: z.string(),
  all_stages: z.array(ClientJobStageSchema),
})
export type ClientJobDetail = z.infer<typeof ClientJobDetailSchema>

export const SessionTypeSchema = z.object({
  id: z.string().uuid(),
  name: z.string(),
  available_days: z.array(z.number()).optional().default([]),
})
export type SessionType = z.infer<typeof SessionTypeSchema>
export const SessionTypeListSchema = z.array(SessionTypeSchema)

export const ClientBookingRequestSlotSchema = z.object({
  session_type_id: z.string().uuid(),
  session_type_name: z.string().nullable(),
  date: z.string(),
  time_slot: z.enum(['morning', 'afternoon', 'evening', 'all_day']),
})
export type ClientBookingRequestSlot = z.infer<typeof ClientBookingRequestSlotSchema>

export const ClientBookingRequestSchema = z.object({
  id: z.string().uuid(),
  preferred_date: z.string(),
  time_slot: z.enum(['morning', 'afternoon', 'evening', 'all_day']).nullable(),
  session_type_name: z.string().nullable(),
  session_slots: z.array(ClientBookingRequestSlotSchema).default([]),
  message: z.string().nullable(),
  status: z.enum(['pending', 'confirmed', 'rejected']),
  admin_notes: z.string().nullable(),
  created_at: z.string(),
})
export type ClientBookingRequest = z.infer<typeof ClientBookingRequestSchema>
export const ClientBookingRequestListSchema = z.array(ClientBookingRequestSchema)
